gateway/qos.go

```diff
@@ -84,11 +84,9 @@ type QoSEndpointCheckGenerator interface {
 	//
 	// GetRequiredQualityChecks returns the set of quality checks required by
 	// the a QoS instance to assess the validity of an endpoint.
-	// The endpoint address is passed here because it allows the QoS instance to
-	// make a decision based on the specific endpoint.
-	// e.g. An EVM-based blockchain service QoS may decide to skip quering an endpoint on
+	// e.g. An EVM-based blockchain service QoS may decide to skip querying an endpoint on
 	// its current block height if it has already failed the chain ID check.
-	GetRequiredQualityChecks(protocol.EndpointAddr) []RequestQoSContext
+	GetRequiredQualityChecks() []RequestQoSContext
```

Comment:
protocol.EndpointAddr should not be dropped from this method: it allows the QoS implementation to decide which checks are needed for a specific endpoint: e.g. no need to perform chainID checks against an endpoint with existing data within the last 10 mnutes.